### ISSUE TITLE
Smear along anti-meridian when using warp resampling other than nearest

### ISSUE THREAD

**Issue by jacott** [OSGeo/gdal#2754]
When converting a `epsg:3857` mosaic to `epsg:2193` selecting a `gdalwarp` resample method other than `nearest` results in a smear down the anti-meridian. It also takes multiple times longer to process. I have tried bilinear, lanczos and cubic which all produce the smear.

## Expected behavior and actual behavior.

Expected:  

![image](https://user-images.githubusercontent.com/94888/86972867-bbb3d900-c1c7-11ea-86a6-5cc7fd861b0b.png)

Actual:

![image](https://user-images.githubusercontent.com/94888/86972921-d1c19980-c1c7-11ea-830f-ede4e46c7b3a.png)

## Steps to reproduce the problem.

[gebco_2019.zip](https://github.com/OSGeo/gdal/files/4893384/gebco_2019.zip) (not the exact images and process used above)

```bash
gdalbuildvrt source.vrt gebco_2019-4-0-9.tiff gebco_2019-4-15-9.tiff 
gdalwarp -t_srs EPSG:2193 -r lanczos source.vrt warp-lanczos.vrt
gdal_translate -of COG -co NUM_THREADS=ALL_CPUS -co COMPRESS=webp -projwin 1890000 6340000 2480000 5900000 -projwin_srs EPSG:2193 warp-lanczos.vrt cog-lanczos.tiff
```

## Operating system

Description:	Ubuntu 20.04 LTS
Linux gkjlinz 5.4.0-40-generic #44-Ubuntu SMP Tue Jun 23 00:01:04 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

## GDAL version and provenance

GDAL 3.2.0dev-1890a2eea4, released 2020/07/04

**Comment by rouault** [comment]
This is a complicated situation... Related to discontinuity of antimerdian of course. The source VRT raster has a huge width due to having a chunk just at its left and side, and nothing in the middle. The discontinuity confuses the warping logic regarding to computing the target resolution (here in automated mode, it would go to 16 meters whereas the source resolution is 306 meters), acquisition of extra pixels around the source window that corresponds to a target window (point addressed by PR #2755) and scaling factors between target pixels and source pixels when doing the resampling (which is the cause of the smearing)

With PR #2755 applied (fixes huge memory allocations that failed on my PC), I manage to get a nice looking result in a few seconds with :
```
gdalwarp -t_srs EPSG:2193 -r lanczos source.vrt tmp.tif -tr 306 306 -overwrite -te 1890000 5900000 2480000 6340000  -co tiled=yes -co compress=zstd  --config GDAL_NUM_THREADS ALL_CPUS -wo XSCALE=1 -wo YSCALE=1 -overwrite
gdal_translate -of COG -co NUM_THREADS=ALL_CPUS -co COMPRESS=webp tmp.tif cog-lanczos.tiff
```
The key is to force the target resolution in the gdalwarp with -tr and force the XSCALE and YSCALE warping options (undocumented hack)

Using a warped VRT as you tried would also work, but is slower:
```
gdalwarp -t_srs EPSG:2193 -r lanczos source.vrt warp-lanczos.vrt  -tr 306 306 -wo XSCALE=1 -wo YSCALE=1 -overwrite
gdal_translate -of COG -co NUM_THREADS=ALL_CPUS -co COMPRESS=webp -projwin 1890000 6340000 2480000 5900000 -projwin_srs EPSG:2193 warp-lanczos.vrt cog-lanczos.tiff
```

Pedantically, if you look at pixel just at the left / right of the antimeridian, you could see that lanczos (or the resampling method) isn't applied since the warping logic will only consider source pixels either at the left or the right of the source VRT, but not both at the same time. For that, you would have to shift one of the source tile in easting by 2 * 20037508.343 , and play with PROJ +over keyword so that warping of longitudes doesn't trigger.

**Comment by rouault** [comment]
With all the fixes now in master, the fastest way to get the result is to run
```
gdalwarp -t_srs EPSG:2193 -r lanczos source.vrt warp-lanczos.tif -te 1890000 5900000 2480000 6340000  -co tiled=yes -co compress=zstd  --config GDAL_NUM_THREADS ALL_CPUS
gdal_translate -of COG -co NUM_THREADS=ALL_CPUS -co COMPRESS=webp  warp-lanczos.tif  cog-lanczos.tiff
```
